falcon/app.py

```diff
@@ -438,7 +444,7 @@ def __call__(self, env, start_response):  # noqa: C901
     def router_options(self):
         return self._router.options
 
-    def add_middleware(self, middleware):
+    def add_middleware(self, middleware: object) -> None:
```

Comment:
we could probably add several protocols for the middleware. sadly is seems that all members of a protocol are required, so I guess we would need to define one for each optional method :(

overall this does not have much impact on the users, so we can likely just do this later on

Reply:
Yeah this is totally out of scope for now unfortunately, my experience with protocols so far was: :cold_sweat:.